spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsMessageDrivenEndpoint.java

```diff
@@ -90,12 +93,63 @@ public void setSessionAcknowledgeMode(String sessionAcknowledgeMode) {
 	}
 
 	@Override
+	public void setOutputChannel(MessageChannel outputChannel) {
+		super.setOutputChannel(outputChannel);
+		this.listener.setRequestChannel(outputChannel);
+	}
+
+	@Override
+	public void setOutputChannelName(String outputChannelName) {
+		super.setOutputChannelName(outputChannelName);
+		this.listener.setRequestChannelName(outputChannelName);
+	}
+
+	@Override
+	public void setErrorChannel(MessageChannel errorChannel) {
+		super.setErrorChannel(errorChannel);
+		this.listener.setErrorChannel(errorChannel);
+	}
+
+	@Override
+	public void setErrorChannelName(String errorChannelName) {
+		super.setErrorChannelName(errorChannelName);
+		this.listener.setErrorChannelName(errorChannelName);
+	}
+
+	@Override
+	public void setSendTimeout(long sendTimeout) {
+		super.setSendTimeout(sendTimeout);
+		this.listener.setRequestTimeout(sendTimeout);
+	}
+
+	@Override
+	public void setShouldTrack(boolean shouldTrack) {
+		super.setShouldTrack(shouldTrack);
+		this.listener.setShouldTrack(shouldTrack);
+	}
+
+	public ChannelPublishingJmsMessageListener getListener() {
+		return this.listener;
+	}
+
+	@Override
+	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+		super.setApplicationContext(applicationContext);
+		this.listener.setBeanFactory(applicationContext);
+	}
+
+	@Override
 	public String getComponentType() {
 		return "jms:message-driven-channel-adapter";
 	}
 
 	@Override
-	protected void onInit() throws Exception {
+	public void afterSingletonsInstantiated() {
+		// skip the output channel requirement test for when the listener is pre-built
```

Comment:
See the super method - when using the XML parser, the listener is set up before injecting - and the setters here are not called.